iris_ugrid/regrid.py

```diff
@@ -0,0 +1,420 @@
+            node_coords[:,1] describes the latitides in degrees
+        * face_node_connectivity
+            A numpy masked array describing the face node connectivity of the
+            mesh. The unmasked points of face_node_connectivity[i] describe
+            which nodes are connected to the i'th face.
+        * node_start_index
+            An integer describing which index is considered the initial index
+            by face_node_connectivity. UGRID supports both 0 based and 1 based
+            indexing, so both must be accounted for here
+
+        Kwargs:
+
+        * elem_start_index
+            An integer describing what index should be considered by ESMF to be
+            the start index for describing its elements. This makes no
+            difference to the regridding calculation and will only affect the
+            intermediate ESMF objects, should the user need access to them.
+        * areas
+            either None or a numpy array describing the areas associated with
+            each face. If None, then ESMF will use its own calculated areas.
+        """
+        self.node_coords = node_coords
+        self.fnc = face_node_connectivity
+        self.nsi = node_start_index
+        self.esi = elem_start_index
+        self.areas = areas
+
+    def _as_esmf_info(self):
+        # ESMF uses a slightly different format to UGRID,
+        # the data must be translated into a form ESMF understands
+        num_node = self.node_coords.shape[0]
+        num_elem = self.fnc.shape[0]
+        nodeId = np.array(range(self.nsi, self.nsi + num_node))
+        nodeCoord = self.node_coords.flatten()
+        nodeOwner = np.zeros([num_node])  # regridding currently serial
+        elemId = np.array(range(self.esi, self.esi + num_elem))
+        elemType = self.fnc.count(axis=1)
+        # Experiments seem to indicate that ESMF is using 0 indexing here
+        elemConn = self.fnc.compressed() - self.nsi
+        result = (
+            num_node,
+            num_elem,
+            nodeId,
+            nodeCoord,
+            nodeOwner,
+            elemId,
+            elemType,
+            elemConn,
+            self.areas,
+        )
+        return result
+
+    def _make_esmf_mesh(self):
+        info = self._as_esmf_info()
+        (
+            num_node,
+            num_elem,
+            nodeId,
+            nodeCoord,
+            nodeOwner,
+            elemId,
+            elemType,
+            elemConn,
+            areas,
+        ) = info
+        # ESMF can handle other dimensionalities, but we are unlikely
+        # to make such a use of ESMF
+        emesh = ESMF.Mesh(
+            parametric_dim=2, spatial_dim=2, coord_sys=ESMF.CoordSys.SPH_DEG
+        )
+
+        emesh.add_nodes(num_node, nodeId, nodeCoord, nodeOwner)
+        emesh.add_elements(
+            num_elem, elemId, elemType, elemConn, element_area=areas
+        )
+        return emesh
+
+    def make_esmf_field(self):
```

Comment:
P.S. _really_ not sure about this idea.  Do consider, but It might be better to leave it as it is.

Reply:
My thinking was that a user could concievably want to control ESMF regridding themselves but still want access to our "translation into ESMF pipeline". This seems like it would be the most sensible method to make public to allow that. With that said, if it turns out to be too much of a commitment to support, there's definitely a case for making it private.

Reply:
Ok, I think that makes sense, especially as you added an account usage to the class docstrings.